diff --git a/tilt/infra.Tiltfile b/tilt/infra.Tiltfile
index 31c04d51..7ec46378 100644
--- a/tilt/infra.Tiltfile
+++ b/tilt/infra.Tiltfile
@@ -242,7 +242,8 @@ local_resource(
         -v ./apps/aabundler/etc:/app/etc/aabundler \
         -e "DEBUG={bundler_debug}" \
         -e "DEBUG_COLORS=true" \
-        0xbigboss/bundler:0.7.1 \
+        -m 200m \
+        docker.io/0xbigboss/bundler:0.7.1-9ae4952 \
         --port 3030 \
         --config /app/etc/aabundler/aabundler.config.json \
         --mnemonic /app/keys/0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 \
